### PR TITLE
chore: dx improvement for codebuild

### PR DESCRIPTION
**Pull request by AaronZyLee**
#### Description of changes

- E2e test
  - The `api_2` is splitted to `api_11` with the most time-consuming case (turn on conflict resolution and then turn off)
  - Fixed `iterative-rollback` tests not catching the exit code issue. In codebuild, the `CtrlC` is not sending the 130 but 2 exit code
- Cleanup
  - Fixed the issue of cleanup job always cleaning all stale resources in the e2e workflow. It should only clean the resources within the batch
  - `stale-resources.json` are correctly generated for each individual account
  - The info of stale resources are also uploaded as artifacts for cleanup job
- Artifact
  - Test artifacts are now correctly uploaded for those run with retries. The `RUN_INDEX` is added in the path. Previously only the second run is uploaded.
  - Also fix the view artifact script to support this change
  - Use `yarn view-test-artifact <code-build-id>` instead of s3 URI to view the artifact of single build

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] Any CDK or CloudFormation parameter changes are called out explicitly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-e2e-tests/src/__tests__/api_2.test.ts [hunk 1]
```diff
@@ -129,43 +127,6 @@ describe('amplify add api (GraphQL)', () => {
     expect(updateResultData.updateNote.note).toEqual(updateInput.input.note);
   });
 
-  it('init a project with conflict detection enabled and toggle disable', async () => {
```
Comment (AaronZyLee) [inline packages/amplify-e2e-tests/src/__tests__/api_2.test.ts:132]: This case takes the majority time of the whole test suites(around 30~40mins). This is moved to a new test file `api_11`

File: scripts/view-test-artifacts.ts [hunk 2]
```diff
@@ -115,7 +115,8 @@ const downloadSingleTestArtifact = async (tempDir: string, artifact: Required<Te
 
   return Promise.all(
     listObjectsResponse.Contents.filter(hasKey).map(({ Key }) => {
-      const filePath = path.join(artifactDownloadPath, Key.split('/').pop()!);
+      const filePath = path.join(artifactDownloadPath, Key.split('/').slice(3).join('/'));
```
Comment (AaronZyLee) [inline scripts/view-test-artifacts.ts:118]: This change is to support the artifact change, which have the run index (0 & 1) in the path

File: shared-scripts.sh [hunk 3]
```diff
@@ -269,7 +269,7 @@ function _cleanupE2EResources {
   echo "Running clean up script"
   build_batch_arn=$(aws codebuild batch-get-builds --ids $CODEBUILD_BUILD_ID | jq -r -c '.builds[0].buildBatchArn')
   echo "Cleanup resources for batch build $build_batch_arn"
-  yarn clean-e2e-resources --buildBatchArn $build_batch_arn
+  yarn clean-e2e-resources buildBatchArn $build_batch_arn
```
Comment (AaronZyLee) [inline shared-scripts.sh:272]: This is to fix the bug that the cleanup job in e2e workflow always cleaning all the stale resources. It should only cleanup the resources generated within the batch

File: packages/amplify-e2e-core/src/categories/auth.ts [hunk 4]
```diff
@@ -127,7 +127,7 @@ interface AddApiOptions {
 
 const defaultOptions: AddApiOptions = {
   apiName: '\r',
-  testingWithLatestCodebase: true,
+  testingWithLatestCodebase: false,
```
Comment (alharris-at) [inline packages/amplify-e2e-core/src/categories/auth.ts:130]: ? Seems like we want to leave this on.

**Review by alharris-at — COMMENTED** [review]
Overall LGTM - just a question on flipping default behavior to 'testNewCode: false'

File: packages/amplify-e2e-core/src/categories/auth.ts (hunk 4 above)
Comment (phani-srikar) [reply packages/amplify-e2e-core/src/categories/auth.ts:130]: This is set to `true` only for migration tests where we install the older CLI version globally and use the locally built binary as the latest. Unless any migration tests are using these functions without explicitly passing in `testingWithLatestCodebase=true`, we should be good.

**Review by phani-srikar — APPROVED** [review]
Have you run e2e on this change atleast to confirm that none of the migration tests are impacted? Otherwise LGTM.

**Review by alharris-at — APPROVED** [review]

File: packages/amplify-e2e-core/src/categories/auth.ts (hunk 4 above)
Comment (AaronZyLee) [reply packages/amplify-e2e-core/src/categories/auth.ts:130]: When it is true, the test consumes the binary installed within the node modules of repo root. In this test the binary installed from local registry will be used and it cannot be consumed by test unless the value is false.

**Comment by AaronZyLee** [comment]
> Have you run e2e on this change atleast to confirm that none of the migration tests are impacted? Otherwise LGTM.

Yes the migration tests all look good

**Pull request merged**